File: 20180713/Largest Number At Least Twice of Others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int idx = 0;
        while(idx < nums.length){
            int i = 0;
            while(i < nums.length && ( i == idx || nums[idx] >= 2 * nums[i])){
                i += 1;
            }
            if (i == nums.length){
                return idx;
            }
            idx += 1;
        }
        return -1;
    }
}

File: 20180713/Largest Number At Least Twice of Others.py
class Solution(object):
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        idx = 0
        while idx < len(nums):
            i = 0
            while i < len(nums) and (i == idx or nums[idx] >= 2 * nums[i]):
                i += 1
            if i == len(nums):
                return idx
            idx += 1
        return -1